src/MongoDB.Driver/ReadPreferenceResolver.cs

```diff
@@ -0,0 +1,42 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver
+{
+    internal static class ReadPreferenceResolver
+    {
+        public static ReadPreference GetEffectiveReadPreference(
+            IClientSessionHandle session,
+            ReadPreference explicitReadPreference,
+            ReadPreference defaultReadPreference)
+        {
+            if (explicitReadPreference != null)
```

Comment:
The spec doth say: 

> The RunCommand method is considered a read operation and MUST use the transaction’s read preference.

So it seems out of spec to use the explicit read preference when in a transaction.

Reply:
Then the spec doth contradict itself, for it also says:

> Users can pass prohibited options to operations in transactions

> For drivers that allow read concern to be passed to a particular operation, this specification says, "If the user supplies an explicit readConcern via a method option, however, drivers MUST apply the readConcern, which will result in a server error." The same principle applies to write concern and read preference. The alternative is to silently ignore the user's read concern (or other options) if it is prohibited with transactions, but this would be a surprising and undetectable deviation from the user's intent.

Reply:
We can argue it out on https://github.com/mongodb/specifications/pull/330